wrangler-api/src/main/java/io/cdap/wrangler/api/ErrorRecordBase.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright © 2017-2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.wrangler.api;
+
+import java.util.List;
+
+/**
+ * Base class for error record that includes the critical fields.
+ */
+public class ErrorRecordBase {
+
+  // Message as to why the row errored.
+  protected final String message;
+  // Code associated with the message.
+  protected final int code;
+  protected final boolean showInWrangler;
+  protected final List<Object> causes;
```

Comment:
I think it would be better for causes to have a defined structure rather than allow any object. Object, for example, would allow somebody to use a mix of primitives, objects, collections, etc. 

This is probably a larger discussion, and the causes don't seem to be used yet so it may be better to remove this and add it back later when there's a more concrete idea of how it will be used.

Reply:
This depends on the frontend actually. Originally I wanted to put Exceptions/Throwables here, but it's probably not useful to hold a stacktrace and all that. Perhaps we should make a common base class that contains some data pointer and a message and then ValidationIssue is an extension of that with a schema pointer?

Reply:
It will probably end up something like that, with an actual Cause class that either allows users to extend it with arbitrary data, or contains some map of properties that users populate, or something like that. Since there are still open questions about what exactly is needed here and how it will be used, this should be added in the future. Anything we add in wrangler-api cannot be removed/changed without deprecating it for the entirety of a major release.